scratch/evaluate_four_elements_traffic.cc

```diff
@@ -27,29 +27,100 @@
 #include "ns3/applications-module.h"
 #include "ns3/packet-sink.h"
 #include "ns3/random-variable-stream.h"
+#include <numeric>
+#include <unordered_map>
 
 using namespace ns3;
 using namespace std;
 
-NS_LOG_COMPONENT_DEFINE ("EvaluateFourElementsTraffic");
+NS_LOG_COMPONENT_DEFINE ("EvaluateFourElementTraffic");
+
+Time delayAccumulator;
+std::vector<double> packetsSize;  /* Vector to save the size of packets */
+std::vector<double> interArrivalTimes;  /* Vector to save the packets inter-arrival times */
+Ptr<GamingStreamingServer> gamingServer;  /* Pointer to the gaming server application */
+Ptr<PacketSink> packetSink;    /* Pointer to the packet sink application */
+uint64_t lastTotalRx = 0;      /* The value of the last total received bytes */
+Time lastPacketTime = Seconds (1.0);
+
+
+void
+ReceivePacket (Ptr<const Packet> packet, const Address & addr)
+{
+  SeqTsHeader header;
+  if (packet->PeekHeader (header) > 0)
+    {
+      Time packetTime = header.GetTs ();
+      delayAccumulator += Simulator::Now () - packetTime;
+      Time different = packetTime - lastPacketTime;
+      interArrivalTimes.push_back (different.GetMilliSeconds ());
+      lastPacketTime = packetTime;
+    }
+  packetsSize.push_back (packet->GetSize ());
+}
+
+void
+CalculateThroughput ()
+{
+  Time now = Simulator::Now ();    /* Return the simulator's virtual time. */
+  double cur = (packetSink->GetTotalRx () - lastTotalRx) * (double) 8 / 1e3;     /* Convert Application RX Packets to MBits. */
+  std::cout << now.GetSeconds () << "s: \t" << cur << " Kbit/s" << std::endl;
+  lastTotalRx = packetSink->GetTotalRx ();
+  Simulator::Schedule (MilliSeconds (100), &CalculateThroughput);
+}
+
+
+void
+GetCdf  (vector<double> empiricalVector)
+{
+  std::unordered_map<double, int> counts;
+  for (auto v : empiricalVector)
+    {
+      ++counts[v];
+    }
+  std::map<double, int> ordered (counts.begin (), counts.end ());
+  double sum = accumulate (ordered.begin (), ordered.end (), 0,
+                           []( int acc, std::pair<int, double> p )
+                           {
+                             return ( acc + p.second );
+                           });
+  double cdf = 0;
+  for (auto it = ordered.begin (); it != ordered.end (); ++it)
+    {
+      cdf += it->second;
+      std::cout << it->first << " " << cdf / sum << std::endl;
+    }
+}
```

Comment:
i'm not sure about the stability and efficiency of this method. Creating a map with double keys sounds a bit crazy to me.
i think that ordering `empiricalVector` and finding its `size` should be enough to print the cdf.
If you want to make your life easier you could even print to a file the values and compute/plot the cdf from a different and easier language, e.g., python.